File: game/Entity.java
package game;

abstract class Entity {
    protected String name;
    protected int health;
    protected int attackPower;

    public Entity(String name, int health, int attackPower) {
        this.name = name;
        this.health = health;
        this.attackPower = attackPower;
    }

    public abstract void attack(Entity target);
}

File: game/Entity.cs
namespace game
{
    public class Entity
    {
        public string name;
        public int health;
        public int attackPower;

        public Entity(string name, int health, int attackPower)
        {
            this.name = name;
            this.health = health;
            this.attackPower = attackPower;
        }

        public virtual void Attack(Entity target)
        {
            // Default attack behavior (if any)
        }
    }
}
